src/main/java/org/jabref/gui/EntryTypeView.java

```diff
@@ -0,0 +1,291 @@
+        this.basePanel = basePanel;
+        this.setTitle(Localization.lang("Select entry type"));
+        //viewModel = new EntryTypeViewModel(basePanel);
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        //ControlHelper.setAction(cancelButton, getDialogPane(), event -> cancelHandle(event));
+        cancelButton.setOnAction(event -> cancelHandle(event));
+
+        //cancelButton.getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(Globals.getKeyPrefs().getKey(KeyBinding.CLOSE), "close");
+        //cancelButton.getActionMap().put("close", cancelAction);
+        //ControlHelper.setAction(generateButton, getDialogPane(), action -> {
+        //    fetcherWorker.run();
+        //});
+        generateButton.setOnAction(action -> {
+            fetcherWorker.run();
+        });
+        comboBox.onActionProperty().addListener(e -> {
+            idTextField.requestFocus();
+            idTextField.selectAll();
+        });
+
+        idTextField.onActionProperty().addListener(event -> fetcherWorker.run());
+
+        if (basePanel.getBibDatabaseContext().isBiblatexMode()) {
+            biblatexPane.setContent(createPane(BiblatexEntryTypes.ALL));
+            vBox.getChildren().remove(bibTexPane);
+            vBox.getChildren().remove(ieeetranPane);
+            List<EntryType> customTypes = EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBLATEX);
+            if (EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBLATEX).isEmpty()) {
+                vBox.getChildren().remove(customPane);
+            } else {
+                customPane.setContent(createPane(customTypes));
+            }
+            //vBox.getChildren().remove(customPane);
+            //bibTexPane.setVisible(false);
+            //ieeetranPane.setVisible(false);
+        } else {
+            bibTexPane.setContent(createPane(BibtexEntryTypes.ALL));
+            ieeetranPane.setContent(createPane(IEEETranEntryTypes.ALL));
+            vBox.getChildren().remove(biblatexPane);
+
+            List<EntryType> customTypes = EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBTEX);
+            if (EntryTypes.getAllCustomTypes(BibDatabaseMode.BIBTEX).isEmpty()) {
+                vBox.getChildren().remove(customPane);
+            } else {
+                customPane.setContent(createPane(customTypes));
+            }
+            //vBox.getChildren().remove(customPane);
+            //biblatexPane.setVisible(false);
+        }
+
+    }
+
+    private GridPane createPane(Collection<? extends EntryType> entries) {
+        GridPane gridpane = new GridPane();
+        gridpane.setPadding(new Insets(4));
+        gridpane.setVgap(4);
+        // row count
+        int row = 0;
+        // col count
+        int col = 0;
+        for (EntryType entryType : entries) {
+            TypeButton entryButton = new TypeButton(entryType.getName(), entryType);
+            entryButton.setOnAction(event -> cancelHandle(event));
```

Comment:
I think, the usage of a custom button class is a bit too complicated. An action like the following should work:
````
entryButton.setOnAction(event -> {
   type.set(entryType);
   close();
});
````
and change `type` to `ObjectProperty`.